homeassistant/components/local_todo/todo.py

```diff
@@ -50,7 +50,10 @@ async def async_setup_entry(
     """Set up the local_todo todo platform."""
 
     store = hass.data[DOMAIN][config_entry.entry_id]
-    ics = await store.async_load()
+    try:
+        ics = await store.async_load()
+    except OSError as err:
+        raise PlatformNotReady() from err
```

Comment:
I'd rather load the first time during config entry setup and raise ConfigEntryNotReady.

Reply:
But this is also ok.

Reply:
This has all been moved, i think following the spirit of your recommendation.